### DISCUSSION TITLE
Default 'Checkt' for checkboxlist

### DISCUSSION THREAD

**Discussion by erikjanwestendorp** [leekelleher/umbraco-contentment#224] [Idea]
First of all what a really awesome package! 😄. 

Is it an idea to add an extra option to the checkbox list configuration that gives the ability to set the default value? Of course I can implement my own IDataListEditor and add the config I want, but i thought it might be a nice feature 😄 

![image](https://user-images.githubusercontent.com/7831614/171006749-ea464ed9-13e5-4f6a-9ab9-08840848979b.png)

**Comment by leekelleher** [comment]
@erikjanwestendorp Thank you for the suggestion. 😀 

To clarify, are you meaning that the configuration would auto-check all of the options in the checkbox list?

If so, the slight difficulty with this is that if content nodes are created programmatically, they'll be an assumption that the property's value will be pre-populated with the options from the data-source. But this quickly becomes complex, _([some background reading on my thoughts on discussion #91](https://github.com/leekelleher/umbraco-contentment/discussions/91#discussioncomment-621301)),_ essentially the data-type configuration isn't the place to set default values.

My current workaround to this are to either use Content Templates (Blueprints) or code, e.g. [`SendingContentNotification`](https://our.umbraco.com/documentation/reference/Notifications/EditorModel-Notifications/).

If I've misunderstood, I'm still open to the idea.

**Comment by erikjanwestendorp** [comment]
@leekelleher Yes that's exactly my suggestions. 

I can imagine it's complicated, and of course I can implement a `SendingContentNotification`.  But when a node is created programmatically, the SendingContentNotification is not used. To fix this I would also need to implement a Saving of Publishing Notification. What I'm actually looking for is the same behavior as an Umbrraco TrueFalse datatype. The 'initial state' is also taken into account when the node is created programmatically.

**Reply by leekelleher** [reply]
Thanks for the clarification.  I've had a look at Umbraco core handles the default state for the `TrueFalse` boolean editor, but I could only find that it sets it when the editor is initially loaded in the backoffice (in the browser), it doesn't appear to set it when nodes are created programmatically. _(If anyone knows if it does and where the code to make that happen is, let me know)._

I'll give the feature some thought. Although I'm still torn between where a node's default property values should be configured. I could add documentation to state that the values wouldn't be automatically set when creating nodes programmatically, that might cover the expectations, but then am I setting a precedence for allowing default values for the other list-editors? It'd get complicated. 🤔
